Replace k8s/install/gce with ref/pub-cloud/gce page

tmjd opened the pull request:
The page in its reduced form is probably better suited to be in our
reference/public-cloud location.
The redirect needs to be fixed, I'm hoping to get some help with that.

Addresses #715
Alternate to #727 
Mostly copied from #388

tmjd commented:
The way the latest redirects are updated will be incompatible with the redirect I've tried to include.

caseydavenport commented on master/reference/public-cloud/gce.md:
```diff
@@ -0,0 +1,42 @@
+>\*IPIP: This rule is required only when using Calico with IPIP encapsulation.
+Keep reading for information on when IPIP is required in GCE.
+
+#### Routing Traffic
```
There are two options here: using GCE cloud routes, or using IPIP.

We should talk about both.

tmjd commented:
I added mention of GCE cloud routes and there is already #414 for adding specifics on how to set that up.
I also added a link from the prior GCE in the k8s section to the GCE in the reference section.

caseydavenport commented on master/reference/public-cloud/gce.md:
```diff
@@ -0,0 +1,44 @@
+
+#### Routing Traffic
+
+To use Calico in GCE, either [ipip encapsulation][IPIP] and NAT outgoing must
```
I think we should split these up into two subsections of "Routing traffic"

and then we should make the answer to this question more apparent in this section:

> This rule is required only when using Calico with IPIP encapsulation. Keep reading for information on when IPIP is required in GCE

caseydavenport replied:
We can add more details on using GCE cloud routes later.

tmjd commented on master/reference/public-cloud/gce.md:
```diff
@@ -0,0 +1,54 @@
+
+Container traffic routing can be enabled by setting [IP-in-IP encapsulation][IPIP]
+and NAT outgoing on the configured Calico IP pools.  Setting these ensures
+that all traffic to and from containers will be sent with IP addresses that
```
I'm not sure about this `Setting these ensures ....` sentence, it might be TMI. :grinning:

caseydavenport replied:
Yeah, I'd err on the side of removing explanatory stuff and keeping it focused on "What do you need to do to make it work"

tmjd commented:
@caseydavenport, @djosborne PTAL

caseydavenport commented on master/reference/public-cloud/gce.md:
```diff
@@ -0,0 +1,54 @@
+
+Traffic routing in GCE can be achieved by utilizing GCE cloud routes.
+Kubernetes GCE cloud provider integration simplifies route configuration
+by enabling Kubernetes to handle creating routes.
```
I'd just add a but of context saying adding ".. and running Calico in policy-only mode"

tmjd commented:
These changes need to be applied to v2.2 before merging.  Are we there? :slightly_smiling_face:

djosborne commented on _data/master/navbars/getting-started.yml:
```diff
@@ -27,8 +27,6 @@ toc:
     - title: AWS
       path: /getting-started/kubernetes/installation/aws
-    - title: GCE
-      path: /getting-started/kubernetes/installation/gce
```
I think this needs to stay?

caseydavenport commented on master/getting-started/kubernetes/installation/gce.md:
```diff
@@ -2,140 +2,5 @@
-[cni-plugin]: https://github.com/projectcalico/cni-plugin
-[coreos-gce]: https://coreos.com/os/docs/latest/booting-on-google-compute-engine.html
-[gcloud-instructions]: https://cloud.google.com/sdk/
+See [this page]({{site.baseurl}}/{{page.version}}/reference/public-cloud/gce)
```
Do we have an idea what the future of this page will be? Will it eventually just be removed, or can we turn it into something like this http://docs.projectcalico.org/v2.1/getting-started/kubernetes/installation/aws?

I think leaving it as a link is OK for now, but ideally we follow up quickly-ish with something similar to the AWS doc.  WDYT?

tmjd replied:
From my talking with Turk I believe we expect those pages to go away as much as possible with only stuff that would be directly needed for k8s.  The reason the page was not dropped altogether was to have a way to redirect to the new page.  (Using a redirect_from would not mesh with our current `latest` redirection.)

caseydavenport replied:
I think I disagree with @djosborne about wanting those pages to go away as much as possible, but that's a discussion we can have later :)

caseydavenport commented:
LGTM - we can deal with follow on actions for the getting started GCE doc in the near future.

tmjd commented:
@djosborne I added that link back in and applied the changes to v2.2, can you merged this please?
